src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -356,6 +356,16 @@
     @Parameter( defaultValue = "${session}", readonly = true, required = true )
     private MavenSession session;
 
+    /**
+     * The core maven model readers/writers are discarding the comments of the pom.xml.
+     * By setting keepCommentsInPom to true the current comments are moved to the flattened pom.xml.
+     * Default value is false (= not re-adding comments).
+     * 
+     * @since 1.2.8
+     */
+    @Parameter( property = "flatten.dependency.keepComments", required = false , defaultValue = "false")
+    private Boolean keepCommentsInPom;
```

Comment:
no need to use `Boolean` but simply use primitive and you can get rid of the method `isKeepCommentsInPom()` and use directly the field.

Reply:
I changed it to primitive boolean.